x-pack/dockerlogbeat/pipereader/reader_test.go

```diff
@@ -5,41 +5,19 @@
 package pipereader
 
 import (
-	"bytes"
-	"encoding/binary"
-	"io/ioutil"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 
-	"github.com/gogo/protobuf/proto"
-
 	"github.com/docker/docker/api/types/plugins/logdriver"
 )
 
 func TestPipeReader(t *testing.T) {
 
-	//setup
-	exampleStruct := &logdriver.LogEntry{
-		Source:   "Test",
-		TimeNano: 0,
-		Line:     []byte("This is a log line"),
-		Partial:  false,
-		PartialLogMetadata: &logdriver.PartialLogEntryMetadata{
-			Last:    false,
-			Id:      "",
-			Ordinal: 0,
-		},
-	}
-	rawBytes, err := proto.Marshal(exampleStruct)
-	assert.NoError(t, err)
-
-	sizeBytes := make([]byte, 4)
-	binary.BigEndian.PutUint32(sizeBytes, uint32(len(rawBytes)))
-	rawBytes = append(sizeBytes, rawBytes...)
+	rawBytes := pipeineMock.CreateTestInput(t)
 
 	// actual test
-	pipeRead, err := NewReaderFromReadCloser(ioutil.NopCloser(bytes.NewReader(rawBytes)))
+	pipeRead, err := NewReaderFromReadCloser(rawBytes)
```

Comment:
Why again do we require the pipeline reader to have an `io.ReadCloser` as input? Could we separate reading + closing better?